x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -65,7 +66,13 @@ public Verifier(Metrics metrics) {
         this.metrics = metrics;
     }
 
-    Collection<Failure> verify(LogicalPlan plan) {
+    /**
+     * Verify that a {@link LogicalPlan} can be executed.
+     *
+     * @param plan The logical plan to be verified
+     * @return a collection of verification failures; empty if and only if the plan is valid
+     */
+    public Collection<Failure> verify(LogicalPlan plan) {
```

Comment:
This method is not meant to be called publicly but instead only by the analyzer (or the tests).
I don't see it called anywhere else in the code either.
Keep the protected visibility.